mcp/utils.go

```diff
@@ -266,6 +266,24 @@ func NewToolResultError(text string) *CallToolResult {
 	}
 }
 
+// NewToolResultErrorWithErr creates a new CallToolResult with an error message.
+// If an error is provided, its details will be appended to the text message.
+// Any errors that originate from the tool SHOULD be reported inside the result object.
+func NewToolResultErrorWithErr(text string, err error) *CallToolResult {
```

Comment:
What do you think about `NewToolResultErrorFromErr` instead? Semantically the same, maybe a bit more clearer.

We could even go `NewToolResultErrorFrom`.